File: Java OOP/Homework05/models/Department.java
package models;

import java.util.ArrayList;

public class Department <T extends Person> {
    private String name;    
    private ArrayList<T> employees = new ArrayList();
    private String departmentType;

    public Department(String name, String departmentType) {
        this.name = name;
        this.departmentType = departmentType;
    }

    public void add(T employee) {
        employees.add(employee);
    }

    public void remove(int index) {
        employees.remove(index);
    }

    public String getName() {
        return this.name;
    }

    public String getDepartmentType() {
        return this.departmentType;
    }

    public void setName(String name) {
        this.name = name;
    }
    
    public ArrayList<T> getEmployees() {
        return this.employees;
    }

    public Person getEmployees(int index) {
        return this.employees.get(index);
    }
}
  

File: Java OOP/Homework07/models/Department.py
from typing import TypeVar, List

T = TypeVar('T', bound='Person')


class Department:
    def __init__(self, name: str, department_type: str):
        self.name = name
        self.department_type = department_type
        self.employees: List[T] = []

    def add(self, employee: T):
        self.employees.append(employee)

    def remove(self, index: int):
        self.employees.pop(index)

    def get_name(self) -> str:
        return self.name

    def get_department_type(self) -> str:
        return self.department_type

    def set_name(self, name: str):
        self.name = name

    def get_employees(self) -> List[T]:
        return self.employees

    def get_employee(self, index: int) -> T:
        return self.employees[index]
